### ISSUE TITLE
IFC Edit Mode menu allows switching to edit mode for a locked object

### ISSUE THREAD

**Issue by kurb70** [IfcOpenShell/IfcOpenShell#5371]
I select an existing room in [awmhhi01a-3_inkl_IfcSpace.ifc.txt](https://github.com/user-attachments/files/16977732/awmhhi01a-3_inkl_IfcSpace.ifc.txt). As soon as I switch to IFC Edit Mode via the menu, I get the message “Geometry changes will be lost”. If I then click on “Discard Changes”, the following error message appears. Surprisingly, the program then appears to be in a strange editing mode.

After clicking on “Discard Changes”, BB should return to a stable state.

![Screenshot 2024-09-12 um 12 33 52](https://github.com/user-attachments/assets/f3138867-7404-4e7b-9b18-2e9c55126bbe)

```
os: Darwin
os_version: Darwin Kernel Version 23.6.0: Mon Jul 29 21:14:30 PDT 2024; root:xnu-10063.141.2~1/RELEASE_ARM64_T6000
python_version: 3.11.7
architecture: ('64bit', '')
machine: arm64
processor: arm
blender_version: 4.2.0
bonsai_version: 0.8.1-alpha240912
bonsai_commit_hash: 35a4d87
last_actions: 
# operator: BIM_OT_load_type_thumbnails
# operator: BIM_OT_override_mode_set_edit
# operator: BIM_OT_override_mode_set_edit
# operator: BIM_OT_override_mode_set_object
# operator: BIM_OT_override_mode_set_object
last_error: Traceback (most recent call last):
  File "/Users/kuba/Library/Application Support/Blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/bim/ifc.py", line 406, in execute_ifc_operator
    result = getattr(operator, "_invoke")(context, None)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/kuba/Library/Application Support/Blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/bim/module/geometry/operator.py", line 1818, in _invoke
    return self.execute(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/kuba/Library/Application Support/Blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/tool/ifc.py", line 249, in execute
    IfcStore.execute_ifc_operator(self, context)
  File "/Users/kuba/Library/Application Support/Blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/bim/ifc.py", line 408, in execute_ifc_operator
    result = getattr(operator, "_execute")(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/kuba/Library/Application Support/Blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/bim/module/geometry/operator.py", line 1721, in _execute
    self.reload_representation(obj)
  File "/Users/kuba/Library/Application Support/Blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/bim/module/geometry/operator.py", line 1731, in reload_representation
    core.switch_representation(
  File "/Users/kuba/Library/Application Support/Blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/core/geometry.py", line 148, in switch_representation
    geometry.change_object_data(obj, new_repr_data, is_global=is_global and not use_immediate_repr)
  File "/Users/kuba/Library/Application Support/Blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/tool/geometry.py", line 53, in change_object_data
    cls.replace_object_data_globally(obj.data, data)
  File "/Users/kuba/Library/Application Support/Blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/tool/geometry.py", line 60, in replace_object_data_globally
    raise Exception("user_remap is not supported for meshes in EDIT mode")
Exception: user_remap is not supported for meshes in EDIT mode

ifc: awmhhi01a-3_inkl_IfcSpace.ifc
schema: IFC4
preprocessor_version: IfcOpenShell 0.7.11-alpha240807
originating_system: BlenderBIM 0.7.11-alpha240807-4fd802b
```

**Comment by Andrej730** [comment]
The issue was that menu allowed switching to edit mode for a locked object leading to that confusing Bonsai state.

We're still establishing workflow with locked objects (spatial objects and grids are locked by default) but currently to edit them you need toggle that lock button in the spatial manager:

![image](https://github.com/user-attachments/assets/724d08ae-fb92-4131-b424-4622254f289e)

**Comment by Moult** [comment]
Yep sorry about that the locking concept is a bit new. It was probably resolved around c5fc86d14851b7f7fd5ac22c5885b45cedf661a8 too
